File: 0162-find-peak-element/solution.java
class Solution {
    public int findPeakElement(int[] nums) {
        int start = 0;
        int end = nums.length - 1;
        while(end >= start){
            int mid = (start + end) / 2;
            if(mid+1 < nums.length && nums[mid+1] > nums[mid]){
                start = mid + 1;
            }
            else if(mid-1 >= 0 && nums[mid-1] > nums[mid]){
                end = mid - 1;
            }
            else{
                return mid;
            }
        }
        return 1;
    }
}

File: 0162-find-peak-element/solution.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        start, end = 0, len(nums) - 1
        while start < end:
            mid = (start + end) // 2
            if nums[mid] < nums[mid + 1]:
                # Peak is to the right
                start = mid + 1
            else:
                # Peak is at mid or to the left
                end = mid
        return start
